crates/ere-zisk/src/lib.rs

```diff
@@ -352,6 +358,42 @@ fn serialize_inputs(inputs: &Input) -> Result<Vec<u8>, bincode::Error> {
     })
 }
 
+/// Deserialize public values as json string sequence, and parse the `RomDigest`
+/// and user set public values as `Vec<u8>`.
+fn deserialize_public_values(public_values: &[u8]) -> Result<(RomDigest, Vec<u8>), CommonError> {
+    let public_values = serde_json::from_slice::<Vec<String>>(public_values)
+        .map_err(CommonError::DeserializePublicValues)?
+        .into_iter()
+        .map(|v| v.parse())
+        .collect::<Result<Vec<u64>, _>>()
+        .map_err(CommonError::ParsePublicValue)?;
+
+    // The public values contain at least the `RomDigest` and the number of user
+    // set public values.
+    if public_values.len() < 5 {
+        return Err(CommonError::InvalidPublicValuesLength(public_values.len()));
+    }
+
+    // The first 4 elements of public values should be ROM digest.
+    let rom_digest = public_values[..4].try_into().unwrap();
+
+    // The 5-th element should be the number of user set public values.
+    let num_user_public_values = public_values[4] as usize;
+
+    // The rest elements should be user set public values and should be `u32`.
+    let public_values = public_values[5..]
+        .iter()
+        .map(|v| Some(u32::try_from(*v).ok()?.to_le_bytes()))
```

Comment:
Was wondering why `u32`, and ended up [here](https://0xpolygonhermez.github.io/zisk/getting_started/writing_programs.html#inputoutput-data). I guess makes sense.

Let's hope eventually they include this function in their SDK, since they might change this in the future and we might have the wrong interpretation of how things are encoded. (Not much we can do for now!)

Reply:
Yes, their guest program only have the `set_output(idx: usize, val: u32)` option to commit public values, so here we downcast it from `u64`.

> Let's hope eventually they include this function in their SDK, since they might change this in the future and we might have the wrong interpretation of how things are encoded.

Make sense!